### DISCUSSION TITLE
Have a Connector field returning one single object instead of an array of 1 element?

### DISCUSSION THREAD

**Discussion by pierpo** [AntlerVC/firetable#1373] [Support / Q&A] [answered]
Hi,

We got to set-up a Connector field, which turned out great to list items from another collection.

Our problem: we need ONE item to be selected. So, we disabled the switch for multiple selection:
<img width="437" alt="image" src="https://github.com/rowyio/rowy/assets/6313316/26329289-acfc-465b-ae26-53f492a9c86a">

But Rowy will still store our element as an array of one single element, where we'd expect it to be a simple item without any array.
Which gives us a data schema that is not really explicit.

```json
{ "teacher": [{"id": "id", "name": "Jean-Michel" }] }
```

instead of 

```json
{ "teachers": {"id": "id", "name": "Jean-Michel" } }
```

Is there a way to not have this array when there is a single element?

I'm aware that there can't be a change on Rowy's side because it would be breaking, but maybe we could have the ability to opt-in with a checkbox or have a custom transformation function before storing the data?

Thanks!

**Comment by hrutik7** [comment, marked as answer]
Hi @pierpo 
You can use a derivative field if they want to store the selected value as a single value instead of array into another field if they want it that way for now

**Reply by pierpo** [reply]
Hey! Many thanks for the workaround, that's very smart. I didn't know that derivative fields existed.
This could solve my problem.

It's still quite convoluted though, it forces to have two fields and having a cloud function running every time we edit a row.
Whereas it could be stored as single field by Rowy directly 😄
